index.js

```diff
@@ -133,9 +133,9 @@ async function send (ctx, path, opts = {}) {
     }
     ctx.set('Cache-Control', directives.join(','))
   }
-  ctx.type = type(path)
   ctx.body = fs.createReadStream(path)
-
+  ctx.type = ctx.res.type ? ctx.res.type : type(path);
```

Comment:
1. `ctx.type` or `ctx.response.type`, not `ctx.res.type`
2. can you show me your use case :)

Reply:
I agree with @nitinbansal1989. I was trying to serve an m4b and let the client operating system to handle de file with the appropriate app. After several debugging hours I found this pull request 😭 Currently the documentation states that:

> You should only use the setHeaders option when you wish to edit the Cache-Control or Last-Modified headers, because doing it before is useless (it's overwritten by send), and doing it after is too late because the headers are already sent.
>
> If you want to edit any other header, simply set them before calling send.

As a quick fix I'm going to manually apply @nitinbansal1989's patch

Reply:
Hi @coderhaoxin 
I'm also agree with @nitinbansal1989 , in this part: "I think if anyone is setting the content type manually, then it should be set to this. "
And also, when using koa-static, we don't have access to "ctx.type", so the content-type on only changed by file extension name.

Reply:
Since this PR is outdated, I'll fix this later.
Or anyone wants to submit a new PR for this :)

Reply:
@coderhaoxin Here you are :)
[#105](https://github.com/koajs/send/pull/105)